desugreet/bot.py

```diff
@@ -4,44 +4,109 @@
+    member_before_entrance = entrance_role in member_before.roles
+    member_after_entrance = entrance_role in member_after.roles
+    # add user to member when the entrance-role is removed
+    if member_before_entrance and not member_after_entrance:
+        # entrance role was removed
+        member_role = discord.utils.get(server.roles, name=member_role_str)
+        if member_role is not None:
+            await member_after.add_roles(member_role)
+        else:
+            print("check your config file, the role does not exist!")
+        # detect when member does not accept DM
+        try:
+            await member_after.send(welcome_msg.format(member_after, server))
+        except discord.errors.Forbidden:
+            await client.get_channel(log_channel_id).send("{0.mention} (closed DM) isch cho".format(member_after))
+        else:
+            await client.get_channel(log_channel_id).send("{0.mention} isch cho".format(member_after))
+        # clean messages in entrance_track_buffer
+        if member_after in entrance_track_buffer.keys():
+            for msg in entrance_track_buffer[member_after]:
+                await msg.delete()
+
+    ## check for boost/de-boosting transition
+    boost_role = discord.utils.get(server.roles, name='Nitro Booster')
+    member_before_boost = boost_role in member_before.roles
+    member_after_boost = boost_role in member_after.roles
+
+    if member_after_boost and not member_before_boost:
+        # booster role was added
+        reaction_emoji = discord.utils.get(server.emojis, name='cirnoSmile')
+        await client.get_channel(log_channel_id).send("{0.mention} het de Server boosted. Danke! {1}".format(member_after, reaction_emoji))
+    if member_before_boost and not member_after_boost:
+        # booster role was removed
+        reaction_emoji = discord.utils.get(server.emojis, name='saddest')
+        await client.get_channel(log_channel_id).send("{0.mention} het ufghört de Server zbooste. {1}".format(member_after, reaction_emoji))
 
 @client.event
 async def on_member_remove(member):
-    await client.get_channel(log_channel_id).send("{0.name} has left".format(member))
+    # suppress the log when the member is of entrance-role
+    server = member.guild
+    entrance_role = entrance_role = discord.utils.get(server.roles, name=entrance_role_str)
+    if entrance_role in member.roles:
+        await client.get_channel(entrance_channel_id).send("{0.name}#{0.discriminator} isch wieder gange".format(member))
+    else:
+        await client.get_channel(log_channel_id).send("{0.name}#{0.discriminator} isch gange".format(member))
 
 
 @client.event
 async def on_message(message):
     if message.author == client.user:
         return
 
+    entrance_channel = client.get_channel(entrance_channel_id)
+    if message.channel is entrance_channel:
+        if message.author in entrance_track_buffer.keys():
+            entrance_track_buffer[message.author].append(message)
```

Comment:
What is the point of keeping the messages in memory exactly? I don't really get the point of it.

Reply:
I record the messages the new entrance-person writes in a list such that I can call delete() method on them.
The list associated with the new user contains the message the bot sends at him and all messages the new user writes in the entrance-channel.
When the user is accepted to Member, delete() method is called on all messages in the list.
After that I probably should remove the dictionary entry. I forgor.

Reply:
we should probably save that to a file as its gone should the bot crash 🤔